Default to the best contest

deboer-tim opened the pull request:
This change allows you to use most commands without specifying a contest id. If the API can connect and determine a "best" contest (e.g. if there is only one running contest or only one today) then it'll output a message and automatically connect to that one.

The algorithm is fairly simple and could likely be improved in the future, but it works for all the cases I could think of. This change covers the following commands: contest, clar, post-clar, and problem. Note that the submit command is *not* complete since it uses the config string directly - I'll submit another PR for that.

nickygerritsen commented on apiTypes_test.go:
```diff
@@ -0,0 +1,51 @@
+package main
```
Time to also run `go test` in the CI 😄 . I will do that soon ™️

deboer-tim committed:
Default to the best contest

This change allows you to use most commands without specifying a contest id. If the API can connect and determine a "best" contest (e.g. if there is only one running contest or only one today) then it'll output a message and automatically connect to that one.

The algorithm is fairly simple and could likely be improved in the future, but it works for all the cases I could think of. This change covers the following commands: contest, clar, post-clar, and problem. Note that the submit command is *not* complete since it uses the config string directly - I'll submit another PR for that.

nickygerritsen commented on commands/root.go:
```diff
@@ -153,3 +176,73 @@ func contestsApi() (interactor.ContestsApi, error) {
+
+	if count == 1 {
+		return best, nil
+	} else if count == 2 {
```
Shouldn’t this be >= 2?

deboer-tim replied:
Yes, fixed.

deboer-tim committed:
Fix >= 2
